### PR TITLE
Merge 1.34.2 into master

### PR DESCRIPTION
**Pull request by vtjnash**
Some commentary:
* This had slightly more problems from the merge than I expected (for example, many new tests inappropriately using types `int` and `ssize_t` instead of `uv_file`, making the post-merge cleanup a bit messier).
* What do we need to do to get mingw64 buildbot running and a `make distcheck` bot?
* The tests from https://github.com/libuv/libuv/pull/849 are failing on my machine due to a mistaken expectation on the reliability of IEEE floating point math on an x87
* The https://github.com/libuv/libuv/pull/1350 tests need to be added to test/test-list.h
* The "remove support for FreeBSD < 10" PR needs to be repeated (https://github.com/libuv/libuv/pull/984)
* The switch from gnu89 to gnu99 PR needs to be repeated (in uv.gyp)
* I wasn't sure the exact status of darwin-proctitle.c, so I just took the file as-is from v1.x—there were too many merge conflicts to sensibly resolve this. Does anyone know if that approach would have lost anything?
* This should all be squashed into the merge commit. Currently that's the actual result of `git merge`, conflict markers and all.
* CI: https://ci.nodejs.org/job/libuv-test-commit/1728/

### CODE REVIEW

**Comment by richardlau** [comment]
>     * What do we need to do to get mingw64 buildbot running and a `make distcheck` bot?
> 

No idea about mingw64 (other than possibly request it on https://github.com/nodejs/build) but there's 
https://github.com/nodejs/build/issues/1910 regarding `distcheck` (I'll add that issue to my list of todos).

We could also test via GitHub actions (it wouldn't give us complete platform coverage but it would be run automatically on pushes and pull requests).

**Comment by vtjnash** [comment]
Ah, nice. I always like when CI runs can apparent trigger bugs in other projects, such as `bash` 😭: https://ci.nodejs.org/job/libuv-test-commit-linux/1792/nodes=ubuntu1404-32/console

Also, it seems like https://github.com/libuv/libuv/pull/1839 should have been merged into v1.x, as it conflicted with b571851e6 in attempting the merge (possibly broken on zos, but the test is skipped on CI). I'll try to reapply that commit. re-CI: https://ci.nodejs.org/job/libuv-test-commit/1729/

> other than possibly request 

It has been requested there (https://github.com/nodejs/build/issues/1581)

File: uv.gyp [hunk 1]
```diff
@@ -92,7 +94,7 @@
           '-Wno-unused-parameter',
           '-Wstrict-prototypes',
         ],
-        'OTHER_CFLAGS': [ '-g', '--std=gnu89', '-pedantic' ],
+        'OTHER_CFLAGS': [ '-g', '--std=gnu89' ],
```
Comment (saghul) [inline uv.gyp]: should this be gnu99, since we switched to c99 in master?

Comment (vtjnash) [reply uv.gyp]: Yeah, unless we want to change master back to c89. I don't really want to combine this change into the merge commit though, so I'll add it to my bulleted list of independent fixups above.

**Comment by vtjnash** [comment]
> As for merging, I never squashed when merging these before.

AFAICT, we've usually dealt with most merge conflicts though before making the merge commit. In this branch, I'm currently preserving exactly what `git merge` gave, then dealt with the merge issues in subsequent commits.

**Comment by saghul** [comment]
> AFAICT, we've usually dealt with most merge conflicts though before making the merge commit.

Yep.



> In this branch, I'm currently preserving exactly what `git merge` gave, then dealt with the merge issues in subsequent commits.

Ah, I see. Ok then, feel free to land!

File: uv.gyp (hunk 1 above)
Comment (saghul) [reply uv.gyp]: Ack, thanks!

**Pull request merged**